statsmodels/tsa/automatic/forecast.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+import pandas as pd
+from scipy.stats import norm
+from statsmodels.tools.decorators import cache_readonly
+
+
+class Forecast:
+    def __init__(self, endog, model, test_sample=0.2, **spec):
+        if type(test_sample) == str:
+            self.endog_training = endog[:test_sample][:-1]
+            self.endog_test = endog[test_sample:]
+        else:
+            if type(test_sample) == float | type(test_sample) == int:
+                if test_sample > 0.0 & test_sample < 1.0:
+                    # here test_sample is containing the number of observations
+                    # to consider for the endog_test
+                    test_sample = int(test_sample * len(endog))
+            self.endog_training = endog[:-test_sample]
+            self.endog_test = endog[-test_sample:]
+        self.endog_training = endog[:test_sample][:-1]
+        self.endog_test = endog[test_sample:]
+        self.model = model(self.endog_training, **spec)
+        self.results = self.model.fit()
+
+    @cache_readonly
+    def nobs_training(self, endog_training):
+        return len(endog_training)
+
+    @cache_readonly
+    def nobs_test(self, endog_test):
+        return len(endog_test)
+
+    @cache_readonly
+    def forecast_val(self):
```

Comment:
For consistency with the state space models, lets call this `forecasts` instead of `forecasts_val`